redhat-cop/rhel-edge-automation-arch
# Bootstrap logic along with BYO

This is the beginning of the culmination of the Helm Migration efforts. This PR contains:

* ArgoCD Chart
* Initialization script to deploy OpenShift GitOps and ArgoCD (ArgoCD is now deployed to a namespace called `rfe-gitops`)
* `user-mgmt` chart which is a superset of the existing `oauth` chart and adds creation of groups and bindings of permissions
* Chart for facilitating the bootstrap logic (Sets up SSH keys and RHSM portal details) and supports the general loading of content (such as raw manifests and secure values etc)
* Values file representing the `byo` overlay from the existing kustomize structure

To test:

1. Add your public/private key to the following locations:
    1.  `charts/bootstrap/files/ssh/image-builder-ssh-private-key`
    2. `charts/bootstrap/files/ssh/image-builder-ssh-public-key`
2. Add RHSM details in the `helm/testing/bootstrap/values.yaml` file
3. Create a values file in `helm/testing/bootstrap/globals.yaml` with the following content to have ArgoCD target this branch for testing:
    ```
    global:
      git:
        url: https://github.com/sabre1041/rhel-edge-automation-arch.git
        ref: bootstrap
    ````
4. Deploy OpenShift GitOps and ArgoCD by running the initialization script at `./setup/init.sh`
5. Bootstrap the environment
    ```
    helm upgrade -i -n rfe-gitops bootstrap charts/bootstrap/ -f helm/testing/bootstrap/values.yaml -f helm/testing/bootstrap/globals.yaml -f helm/testing/byo/byo.yaml
    ````
6. Verify
    1. RFE identity provider has been added enabling htpasswd authentication
    2. Applications deployed in ArgoCD instance in `rfe-gitops` namespace

There are still a few outstanding items, but this would be a HUGE milestone!
